test/remoting.integration.js

```diff
@@ -238,5 +208,154 @@ describe('remoting - integration', function() {
+      ];
+
+      // TODO: update the doc for list of methods accordingly
+      // https://docs.strongloop.com/display/public/LB/Exposing+models+over+REST
+      expect(methods).to.include.members(expectedMethods);
+    });
+  });
+});
+
+describe('With model.settings.replaceOnPUT false' +
+  'set to false', function() {
+  lt.beforeEach.withApp(app);
+  lt.beforeEach.givenModel('storeWithreplaceOnPUTfalse');
+  afterEach(function(done) {
+    this.app.models.storeWithreplaceOnPUTfalse.destroyAll(done);
+  });
+
+  it('should have expected remote methods',
+  function() {
+    var storeClass = findClass('storeWithreplaceOnPUTfalse');
+    var methods = storeClass.methods
+      .filter(function(m) {
+        return m.name.indexOf('__') === -1;
+      })
+      .map(function(m) {
+        return formatMethod(m);
+      });
+
+    var expectedMethods = [
+      'create(data:object):storeWithreplaceOnPUTfalse POST /stores2',
+      'upsert(data:object):storeWithreplaceOnPUTfalse PATCH /stores2',
+//        'upsert(data:object):storeWithreplaceOnPUTfalse PUT /stores2',
+      'replaceOrCreate(data:object):storeWithreplaceOnPUTfalse POST /stores2/replaceOrCreate',
+      'exists(id:any):boolean GET /stores2/:id/exists',
+      'findById(id:any,filter:object):storeWithreplaceOnPUTfalse GET /stores2/:id',
+      'replaceById(id:any,data:object):storeWithreplaceOnPUTfalse POST /stores2/:id/replace',
+      'find(filter:object):storeWithreplaceOnPUTfalse GET /stores2',
+      'findOne(filter:object):storeWithreplaceOnPUTfalse GET /stores2/findOne',
+      'updateAll(where:object,data:object):object POST /stores2/update',
+      'deleteById(id:any):object DELETE /stores2/:id',
+      'count(where:object):number GET /stores2/count',
+      'prototype.updateAttributes(data:object):storeWithreplaceOnPUTfalse PATCH /stores2/:id',
+//        'prototype.updateAttributes(data:object):storeWithreplaceOnPUTfalse PUT /stores2/:id',
+      'createChangeStream(options:object):ReadableStream POST /stores2/change-stream',
+    ];
+
+    // The list of methods is from docs:
+    // https://docs.strongloop.com/display/public/LB/Exposing+models+over+REST
+    expect(methods).to.include.members(expectedMethods);
   });
 });
+
+describe('With model.settings.replaceOnPUT true' +
+  'set to true', function() {
+  lt.beforeEach.withApp(app);
+  lt.beforeEach.givenModel('storeWithReplaceOnPUTtrue');
+  afterEach(function(done) {
+    this.app.models.storeWithReplaceOnPUTtrue.destroyAll(done);
+  });
+
+  it('should have expected remote methods',
+  function() {
+    var storeClass = findClass('storeWithReplaceOnPUTtrue');
+    var methods = storeClass.methods
+      .filter(function(m) {
+        return m.name.indexOf('__') === -1;
+      })
+      .map(function(m) {
+        return formatMethod(m);
+      });
+
+    var expectedMethods = [
+      'create(data:object):storeWithReplaceOnPUTtrue POST /stores3',
```

Comment:
Can we use more descriptive URL paths please? E.g. `/stores-default`, `/stores-replacing`, `/stores-updating`.